frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -129,6 +140,8 @@ class Dashboard extends Component {
     main.addEventListener("resize", this.throttleParameterWidgetStickiness, {
       passive: true,
     });
+
```

Comment:
Would there be a better way to do this? I feel like writing `window.addEventListener` in a bunch of components might seem a little copy-pastey for my liking, not sure if we can have a hook or something that we can use across the class and functional components that the components in this epic are going to use

Reply:
We could use `useBeforeUnload` hook from `react-use` in `DashboardApp` component, which is a functional component.